models/sales/intermediate/int_hubspot__deals_enhanced.sql

```diff
@@ -45,11 +65,20 @@ with deals as (
     left join pipeline_stages 
         on deals.deal_pipeline_stage_id = pipeline_stages.deal_pipeline_stage_id
 
+    {% if var('hubspot_merged_deal_enabled', true) %}
+    left join aggregate_merged_deals
+        on deals.deal_id = aggregate_merged_deals.deal_id
+    {% endif %}
+
     {% if var('hubspot_owner_enabled', true) %}
     left join owners 
         on deals.owner_id = owners.owner_id
     {% endif %}
-)
+
+    {% if var('hubspot_merged_deal_enabled', true) %}
+    where deals.deal_id not in (select merged_deal_id from merged_deals) -- remove deals that have been merged
+    {% endif %}
```

Comment:
Righ,t I was thinking through options here since I know our packages barely use subqueries < CTEs. This was the most straightforward that I came to, but I know subqueries aren't efficient to be used in Where clauses. What do you think about the following?
```

    left join merged_deals
        on deals.deal_id = merged_deals.merged_deal_id
    where merged_deals.merged_deal_id is null
```
    
We want to only have deal_id's from the deal CTE that don't have a corresponding merged_deal_id from merged_deals. This way is less straightforward though